optimum/neuron/modeling_diffusion.py

```diff
@@ -521,7 +522,8 @@ def _from_pretrained(
             if file_paths[1].is_file():
                 model_config = DiffusersPretrainedConfig.from_json_file(file_paths[1])
                 configs[name] = model_config
-                neuron_configs[name] = cls._neuron_config_init(model_config)
+                model_type = "clip-text-with-projection" if name == "text_encoder_2" else None
+                neuron_configs[name] = cls._neuron_config_init(model_config, model_type=model_type)
```

Comment:
Thanks @neo, good catch. I actually put the wrong model type for text encoder 2 when I added the sdxl support, and then just patch the model type when reconstructing the neuron config. The right way to do it is just fix the model type! 

Thxxx that's super super helpful as suggestion! 💯

Reply:
glad it works this way 😆 I didn't really know if it was also coupled/connected with other things

because I saw [here](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/text_encoder_2/config.json#L16) has `clip_text_model` and wasn't sure if it's the same thing